packages/dashboard-frontend/src/services/workspace-client/devworkspace/devWorkspaceClient.ts

```diff
@@ -55,29 +65,73 @@ export const DEVWORKSPACE_METADATA_ANNOTATION = 'dw.metadata.annotations';
  */
 @injectable()
 export class DevWorkspaceClient extends WorkspaceClient {
+  private subscriber: Subscriber | undefined;
   private previousItems: Map<string, Map<string, IStatusUpdate>>;
   private readonly maxStatusAttempts: number;
   private lastDevWorkspaceLog: Map<string, string>;
-  private pluginRegistryUrlEnvName: string;
-  private pluginRegistryInternalUrlEnvName: string;
-  private dashboardUrlEnvName: string;
+  private readonly pluginRegistryUrlEnvName: string;
+  private readonly pluginRegistryInternalUrlEnvName: string;
+  private readonly dashboardUrlEnvName: string;
   private readonly websocketClient: WebsocketClient;
-  private intervalId: number | undefined;
+  private webSocketEventEmitter: EventEmitter;
+  private readonly webSocketEventName: string;
+  private readonly _failingWebSockets: string[];
+  private readonly showAlert: (alert: AlertItem) => void;
 
-  constructor(@inject(KeycloakSetupService) keycloakSetupService: KeycloakSetupService) {
+  constructor(@inject(KeycloakSetupService) keycloakSetupService: KeycloakSetupService,
+              @inject(AppAlerts) appAlerts: AppAlerts) {
     super(keycloakSetupService);
     this.previousItems = new Map();
     this.maxStatusAttempts = 10;
     this.lastDevWorkspaceLog = new Map();
     this.pluginRegistryUrlEnvName = 'CHE_PLUGIN_REGISTRY_URL';
     this.pluginRegistryInternalUrlEnvName = 'CHE_PLUGIN_REGISTRY_INTERNAL_URL';
     this.dashboardUrlEnvName = 'CHE_DASHBOARD_URL';
+    this.webSocketEventEmitter = new EventEmitter();
+    this.webSocketEventName = 'websocketClose';
+    this._failingWebSockets = [];
+
+    this.showAlert = (alert: AlertItem) => appAlerts.showAlert(alert);
+
+    this.websocketClient = new WebsocketClient({
+      onDidWebSocketFailing: (websocketContext: string) => {
+        this._failingWebSockets.push(websocketContext);
+        this.webSocketEventEmitter.emit(this.webSocketEventName);
+      },
+      onDidWebSocketOpen: (websocketContext: string) => {
+        const index = this._failingWebSockets.indexOf(websocketContext);
+        if (index > -1) {
+          this._failingWebSockets.splice(index, 1);
+          this.webSocketEventEmitter.emit(this.webSocketEventName);
+        }
+        this.subscribe().catch(e => {
+          const key = 'websocket-subscribe-error';
+          const title = `Websocket '${websocketContext}' subscribe Error: ${e}`;
+          this.showAlert({ key, variant: AlertVariant.danger, title });
+        });
+      },
+      onDidWebSocketClose: (event: CloseEvent) => {
+        if(event.code !== 1011 && event.reason) {
+          const key = `websocket-close-code-${event.code}`;
+          this.showAlert({ key, variant: AlertVariant.warning, title: event.reason });
```

Comment:
I assume it's this piece of code that shows that warning 
![Screenshot_20210908_143625](https://user-images.githubusercontent.com/5887312/132502728-6b9034e1-5e5a-4b70-9b26-57e66cda5b47.png)
you can reproduce it by scaling dashboard to 0.
I believe we need to add context there, like
```suggestion
          this.showAlert({ key, variant: AlertVariant.warning, title: 'Failed to establish WebSocket to server: ' +event.reason });
```

Reply:
I have fixed it